Makefile

```diff
@@ -31,6 +34,30 @@ test: ## Run unit tests.
 test-e2e: ## Run E2E tests.
 	go test $(GO_TEST_ARGS) -tags=e2e -coverprofile=coverage.out ./internal/e2e
 
+# Version should match with whatever we consume in sources (check the go.mod).
+SETUP_ENVTEST 			:= go run sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
+ENVTEST_BIN_DIR 		:= $(shell pwd)/bin
+
+# The kubernetes version to use with envtest. Overridable when invoking make.
+# E.g. make ENVTEST_KUBE_VERSION=1.24 test-integration
+ENVTEST_KUBE_VERSION 	?= 1.25
+
+.PHONY: setup-envtest
+setup-envtest:
+	@echo Installing Kubernetes $(ENVTEST_KUBE_VERSION) binaries into $(ENVTEST_BIN_DIR); \
+	$(SETUP_ENVTEST) use --bin-dir $(ENVTEST_BIN_DIR) $(ENVTEST_KUBE_VERSION)
+
+# Integration tests are located next to unit test. This recipe will search the code base for
+# files including the "//go:build integration" build tag and build them into the test binary.
```

Comment:
What are your thoughts on Peter Bourgon? https://peter.bourgon.org/blog/2021/04/02/dont-use-build-tags-for-integration-tests.html
Sorry, I don't mean to throw a wrench into this work you've done.

Reply:
I've not seen that article post before, but it was interesting. 

I'm not opposed to not using build tags; I've felt the problems the article talks about. I'm also not sure how bothered I am because the Makefile target documentation needs to clarify expectations irrespective of the approach used. The other benefit is the explicit log message which could be a signal for the ignorant to investigate further - we lack that entirely with the tags approach.

That said, I'm not a fan of skipping the test based on missing config. I'd rather have a single toggle "RunSuiteA" and the test fails for any missing required config. I feel like that's much more intentional.

Reply:
what does a toggle look? Also, im good with this as is. We have some discussion items but nothing so pressing as to block this.

Reply:
The toggle would be the same as in that article. It would replace the `is config present`. For example:

```go
func TestIntegration(t *testing.T) {
	if _, ok := os.LookupEnv(EnableIntegrationTestsEnvName); !ok {
		// This message can be more specific to the integration test suite.
		t.Skipf("To enable the integration test suite set %s=true", EnableIntegrationTestEnvName)
	}

	// Check if integration test configuration is set.
	if integrationTestConfigurationIsNotSet() {
		t.Fatal("Missing required config")
	}
}
```

I prefer this because it accommodated multiple configuration variables and feels more intentional in terms of 'turning on the test suite'.

Reply:
There's quite a few PRs open with various changes so I'm going to stick with this approach for the PR but we can continue the discussion.